client/data/home/use-home-layout-query.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import { useQuery, UseQueryResult } from 'react-query';
+
+/**
+ * Internal dependencies
+ */
+import wpcom from 'calypso/lib/wp';
+import { getHomeLayoutFlags } from './home-layout-flags';
+
+interface Options {
+	fetchOnMount?: boolean;
+}
+
+const useHomeLayoutQuery = (
+	siteId: number | null,
+	{ fetchOnMount = false }: Options = {}
+): UseQueryResult => {
+	const { isDev, forcedView } = getHomeLayoutFlags();
+
+	return useQuery(
+		[ 'home-layout', siteId ],
+		() =>
+			wpcom.req.get(
+				{
+					path: `/sites/${ siteId }/home/layout`,
+					apiNamespace: 'wpcom/v2',
+				},
+				{
+					...( isDev && { dev: true } ),
+					...( forcedView && { view: forcedView } ),
+				}
```

Comment:
If `getHomeLayoutFlags` returns `{ dev: undefined }`, the `dev` key will be omitted from the actual query string. That way, you could directly pass the return value without structuring, destructuring, and structuring objects again.